src/main/java/com/reimbursements/controllers/UserController.java

```diff
@@ -0,0 +1,63 @@
+package com.reimbursements.controllers;
+
+import java.security.Principal;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.reimbursements.models.User;
+import com.reimbursements.requests.CreateAuthorityRequest;
+import com.reimbursements.requests.CreateUserRequest;
+import com.reimbursements.services.AuthorityService;
+import com.reimbursements.services.UserService;
+
+@RestController
+@RequestMapping(value = "/users", method = {RequestMethod.GET, RequestMethod.POST})
+@CrossOrigin(origins = "http://localhost:4200", allowCredentials = "true")
+public class UserController {
+	
+private UserService userService;
+private AuthorityService authorityService;
+	
+	@Autowired
+	public UserController(UserService userService, AuthorityService authorityService) {
+		super();
+		this.userService = userService;
+		this.authorityService = authorityService;
+	}
+	
+	@Bean 
+	public BCryptPasswordEncoder bCryptPasswordEncoder() {
+		return new BCryptPasswordEncoder();
+	}

```

Comment:
Please don't add bean configuration to controllers, services, components. `@Bean`s should be added to class annotated with `@Configuration`. It's how the spring initializes context, it would be easier to have it in one place.